### PR TITLE
Add session storage support with function option

### PR DESCRIPTION
**Pull request by ekwoka**
Great work dude. I've never done a pull request before, so please let me know if I did anything wrong in this process, just edited the file on github. Worked on a separate thing I was playing with. Seems this could be the cleanest way to add session storage support without breaking anything for anyone else and keeping it DRY.

### CODE REVIEW

**Comment by ekwoka** [comment]
forgot to write how to use:

Alpine.persistedStore('this is in', 'localStorage');
Alpine.persistedStore('this is in', 'sessionStorage', 0);

File: src/index.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ export default function (Alpine) {
 
     window.__ferns = {}
 
-    Alpine.persistedStore = function (name, value) {
-        let stored = localStorage.getItem(`__fern_${name}`)
+    Alpine.persistedStore = function (name, value, type = 1) {
+        let storage = type ? localStorage : sessionStorage
```
Comment (ryangjchandler) [inline src/index.js:9]: I don't think using a numeric flag / `bool` here really makes sense. Why not just allow the user to pass the storage object through instead?

File: src/index.js [hunk 2]
```diff
@@ -5,8 +5,9 @@ export default function (Alpine) {
 
     window.__ferns = {}
 
-    Alpine.persistedStore = function (name, value) {
-        let stored = localStorage.getItem(`__fern_${name}`)
+    Alpine.persistedStore = function (name, value, type = 1) {
```
Comment (ryangjchandler) [inline src/index.js:8]:
```suggestion
    Alpine.persistedStore = function (name, value, storage = window.localStorage) {
```

File: src/index.js (hunk 1 above)
Comment (ryangjchandler) [reply src/index.js:9]:
```suggestion
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (ekwoka) [reply src/index.js:8]: Ooh smart!

**Comment by ryangjchandler** [comment]
@ekwoka Can you confirm that these changes work?

**Comment by ekwoka** [comment]
> @ekwoka Can you confirm that these changes work?

@ryangjchandler This works brilliantly

**Comment by ryangjchandler** [comment]
@ekwoka Could you also update the documentation?

File: README.md [hunk 3]
```diff
@@ -62,6 +62,13 @@ Creating a `persistedStore` will register a normal store with Alpine and keep tr
 </div>
 ```
 
+To use `sessionStorage`, or another custom storage that uses the same syntax, in place of `localStorage`, simply pass an optional third argument containing the storage object when creating the store.
```
Comment (ryangjchandler) [inline README.md:65]:
```suggestion
To use `sessionStorage`, or another custom storage that has an identical API, simply pass an optional third argument containing the storage object when creating the store.
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

**Comment by ekwoka** [comment]
@ryangjchandler done. Should I do the build as well? Is that normal?

**Comment by ryangjchandler** [comment]
Yes please, if you could build and push those files.

**Comment by ekwoka** [comment]
@ryangjchandler 

Done. I guess you didn't see that I built it a while ago haha

**Pull request closed without merging**